apps/server/src/shared/testing/test-request.ts

```diff
@@ -0,0 +1,156 @@
+import { INestApplication } from '@nestjs/common';
+import { Account } from '@shared/domain';
+import supertest from 'supertest';
+import { defaultTestPassword } from './factory/account.factory';
+
+interface AuthenticationResponse {
+	accessToken: string;
+}
+
+enum Header {
+	accept = 'accept',
+	json = 'application/json',
+	authorization = 'authorization',
+}
+
+enum Jwt {
+	invalid = 'invalidJwt',
+	prefix = 'Bearer',
+	loginUri = '/authentication/local',
+	responseKey = 'accessToken',
+	errorMessage = 'TestRequest: Can not cast to local AutenticationResponse:',
+}
+
+export class TestRequest {
+	app: INestApplication;
```

Comment:
Wir wollen doch jetzt vor alle Props die access modifiers schreiben, oder?

Reply:
Und sollten die nicht auch private readonly sein?

Reply:
Ich glaube das geht nicht. Wenn ich oben readonly dazu packe, kann ich sie im constructor nicht mehr belegen. Man könnte
this.app = Object.freeze(app); machen, allerdings würde man damit entweder die app fix machen, oder müsste vorher eine Kopie erzeugen.

Reply:
Das dürfte eigentlich nicht sein. readonly verhindert doch nur ein Assignment außerhalb des Constructors. Siehe: https://www.typescriptlang.org/docs/handbook/2/classes.html#readonly

Reply:
Ah ok, solange man das property noch nicht initalisiert hat geht das. 
Ich hatte das bisher nur in Fällen mit default propbiert gehabt.
readonly xyz = ''  und dann im constructor belegen geht dann nicht mehr.